### DISCUSSION TITLE
Add support for digital meters

### DISCUSSION THREAD

**Discussion by NTohan** [jomjol/AI-on-the-edge-device#3141] [General]
### The Feature

Dear @jomjol,

Thank you for the great initiative, your solution is something we all can leverage in Germany **with minimal efforts**. We are in the transition phase towards more digitalization, but to support existing analog devices is also a very important point. This is exactly where your solution is very helpful. 

Why?
I believe your solution will be also very much helpful in semi-smart meters, where there is a display with digital numbers but yet an additional adapter is needed to collect data from the meter. 

What?
Have you tested your solution on digital Power Meters? I believe reading values from a digital LED can be different than from analog meters because of different gaze?
For instance, https://www.efr.de/wp-content/uploads/2022/06/Produktinformation_EFR_SGM-C4_EN_WEB.pdf

Benefits: 
- Adding support for digital meters can reduce the cost to buy expensive adapters from the manufacturers.
- Any additional adapter to be integrated needs to be installed by the grid provider, i.e., additional effort and cost for each visit. 
- I am afraid, adapter from the manufacturers will also lead to more closed solutions, for instance, missing home assistant integration as a starter. 

Request:
Are the digital meters already supported? Is yes, can you please share the documentation or please add support for the digital power meters? 

Vielen Dank und Grüße,
Navneet

**Comment by SybexX** [comment]
https://jomjol.github.io/AI-on-the-edge-device-docs/
https://jomjol.github.io/neural-network-digital-counter-readout/

This project may also be of interest to you:
https://github.com/arendst/Tasmota
search e.g. on ebay to tasmota lesekopf, hichi lesekopf or tasmota stromzähler wifi .......

**Comment by NTohan** [comment]
@SybexX 

https://github.com/arendst/Tasmota is more relevant for my use case. Also, found a [HW sold on amazon.de](https://www.amazon.de/Hichi-Lesekopf-Stromz%C3%A4hler-optisch-auslesen/dp/B0D6K34LNB/ref=asc_df_B0D6K34LNB/?tag=googshopde-21&linkCode=df0&hvadid=700335351821&hvpos=&hvnetw=g&hvrand=14297503286442712500&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9042418&hvtargid=pla-2372096699447&mcid=f8ab55d20ce03fd9aff037a38c56e288&th=1&gad_source=1) with pre-installed Tasmota firmware. I will definitely give it a try. Thanks for the hint. 

> https://jomjol.github.io/neural-network-digital-counter-readout/

This is an interesting link, I do see samples of digital counter readouts for all the digits. Does that mean a digital display like below is already supported by [AI-on-the-edge-device](https://github.com/jomjol/AI-on-the-edge-device), can you please confirm?

![image](https://github.com/user-attachments/assets/19f1b382-3c19-4131-8b7b-901670192c68)


Thank you.

**Comment by sweti3** [comment]
I would also like to push this! I have a crazy usecase: My PV Inverter also has a display. I would like to read the value every X seconds. Sadly the inverter is a chinese model, so no real LAN or any interface. But a display! I would really love to use that one ... Is this possible? Same question as @NTohan

**Comment by SybexX** [comment]
@NTohan yes, a digital display is also supported. 
For a digital display you just have to deactivate the analog ROIs and only use digital ROIs.
The problem with many digital displays is, that they change the displayed values ​​at a certain interval (in your case between 1.8.1 and 2.8.1).

Here are some examples where you can use it https://github.com/jomjol/AI-on-the-edge-device/discussions/2500 https://github.com/jomjol/AI-on-the-edge-device/discussions/3020 https://github.com/jomjol/AI-on-the-edge-device/issues/2954#issuecomment-2131458653

**Reply by leneffets** [reply]
This would be my usecase. I have a "Smart Power Meter" - but the IR Interface is encrypted and the supplier won't give me the data. The display cycles through different values at set intervals. Can I configure such an interval here, as explained in the documentation? (https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#section-autotimer)

**Reply by caco3** [reply]
> Can I configure such an interval here

This will not work as the rounds of the meter and the ESP will drift apart sooner than later.
There is currently no functionality to detect the mode the meter is in. All you can do is to disable neg rate and have a high interval and use tight values for rate to high.

What would work is this: https://github.com/jomjol/AI-on-the-edge-device/issues/1936, how ever it would be quite some work to implement it and nobody worked on that yet.
